app/scripts/modules/core/src/pipeline/config/triggers/trigger.directive.js

```diff
@@ -36,14 +35,13 @@ module.exports = angular.module('spinnaker.core.pipeline.config.trigger.triggerD
       $scope.pipeline.triggers.splice(triggerIndex, 1);
     };
 
-    this.artifactFilter = function(artifact) {
-      // Ensure suggested artifacts have a name and type
-      return artifact.type && artifact.name;
-    };
-
     this.summarizeExpectedArtifact = function(expected) {
-      const fieldSummaries = expected.fields.filter(field => field.fieldType === FieldType.MustMatch).map(field => field.fieldName + ':' + field.value);
-      return fieldSummaries.join(', ');
+      const artifact = expected.matchArtifact;
+      return Object.keys(artifact)
+        .filter((k) => !k.startsWith('$'))
```

Comment:
I think you can get a $$ here too - just `angular.shallowCopy` the artifact first.

Reply:
`angular.shallowCopy` sounds better - doesn't that `startsWith` check catch `$$` as well?

Reply:
Oh definitely, I just didn't read right 🙃